tests/expand_template/BUILD

```diff
@@ -14,21 +14,36 @@
 
 # This package aids testing the 'diff_test' rule.
 
+load("//rules:common_settings.bzl", "string_flag")
 load("//rules:expand_template.bzl", "expand_template")
 
 package(
     default_applicable_licenses = ["//:license"],
     default_testonly = 1,
 )
 
+string_flag(
+    name = "my_string_flag",
+    build_setting_default = "VAR",
+    make_variable = "SOMEVAR",
+)
+
 expand_template(
     name = "filled_template",
     out = "foo/test.yaml",
+    data = [
+        ":version",
+    ],
     substitutions = {
         "@name@": "test",
         "@version@": "1.1.1",
+        "@path@": "$(rlocationpath :version)",
+        "@toolchain@": "prefix$(SOMEVAR)suffix",
     },
     template = "test.tpl.yaml",
+    toolchains = [
+        ":my_string_flag",
```

Comment:
I didn't realise we can commit such attrocities as putting a flag under toolchains. I've got mixed feeling though, becuase it looks like a nice and elegant interface.

Reply:
yea, idk if it's great or not but at least in this case it means we don't have to introduce another custom thing just for tests